changelog/592.breaking.rst

```diff
@@ -0,0 +1 @@
+Update `~ndcube.NDCubeABC`, `~ndcube.extra_coords.ExtraCoordsABC` and `~ndcube.GlobalCoordsABC` in line with official NDCube 2 API definition in SEP.
```

Comment:
@Cadair I labelled this a breaking change as I've changed the inheritance of `NDCubeABC` from `NDData` to `NDDataBase` but perhaps this is over cautious?  Should this actually be a `trivial` changelog entry?

Reply:
Changed type to `feature` as `GlobalCoordsABC` didn't previously exist and changes to `NDCubeABC` and `GlobalCoordsABC` unlikely to cause breaking changes as these were previously for internal use and not easy to import.